### PR TITLE
Djoudi patch 1

### PR DESCRIPTION
**Pull request by djoudi**

### CODE REVIEW

**Comment by djoudi** [comment]
@xabbuh  check all in this pull please

**Comment by xabbuh** [comment]
It looks like you accidentally changed the line ending style of all files from Unix to Windows line endings. Please revert that.

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
 /composer.lock
 /phpunit.xml
-/vendor
+/vendor/
+/.idea/
```
Comment (xabbuh) [inline .gitignore]: there's no need for these changes, please revert them

Comment (djoudi) [reply .gitignore]: Ok now check it

**Comment by xabbuh** [comment]
I am sorry, but the PR still contains too many unrelated changes. Just take a look at the diff view.

**Pull request closed without merging**